src/Controller/HomeController.php

```diff
@@ -3,52 +3,94 @@
 namespace Controller;
 
 use Model\Manager\CategoryManager;
+use Model\Manager\MemberManager;
 use Model\Manager\PostManager;
 use Twig_Environment;
 
 class HomeController extends Controller
 {
-    /**
-     * @var PostManager
-     */
     protected $postManager;
-    /**
-     * @var CategoryManager
-     */
     protected $categoryManager;
+    private $memberManager;
 
     const VIEW_HOME = 'home/home.twig';
 
     public function __construct(
                                 PostManager $postManager,
                                 CategoryManager $categoryManager,
+                                MemberManager $memberManager,
                                 Twig_Environment $twig
     )
     {
         parent::__construct($twig);
         $this->postManager = $postManager;
         $this->categoryManager = $categoryManager;
+        $this->memberManager = $memberManager;
     }
 
     /**
      * Show the home page
      *
+     * @param string|null $message
+     * @throws \Application\Exception\BlogException
      * @throws \Twig_Error_Loader
      * @throws \Twig_Error_Runtime
      * @throws \Twig_Error_Syntax
      */
-    public function showHome()
+    public function showHome(?string $message = null)
     {
         $categories = $this->categoryManager->getAll();
         $postsByCategory = [];
 
         foreach ($categories as $category) {
             $postsByCategory[$category->getId()] = $this->postManager->getPostsOfACategory($category->getId());
+            // Format creation dates
+            foreach ($postsByCategory[$category->getId()] as $post) {
+                $post->setCreationDate(self::formatDate($post->getCreationDate()));
+                BlogController::decodePostExcerpt($post);
+                if ($post->getLastModificationDate()) {
+                    $post->setLastModificationDate(self::formatDate($post->getLastModificationDate()));
+                }
+            }
         }
 
         self::render(self::VIEW_HOME, [
             'categories' => $categories,
-            'postsByCategory' => $postsByCategory
+            'postsByCategory' => $postsByCategory,
+            'numberOfPosts' => 5,
```

Comment:
Why do you fetch ALL the posts if you don't want to display all of them?
Don't filter in the twig, add a param to your `getPostsOfACategory` adding a limit to your SQL query.
Your performances will be better :)

Reply:
This will be corrected in the *Paging* pull request.